bearer to 1.31.1 panic: runtime error: invalid memory address or nil pointer dereference

<!--- Provide a general summary of the issue in the Title above -->

Updating from 1.31.0 started throwing this error in github actions

## Description & Reproduction

<!--- 

Please provide the command used that causes the error 
if possible link to code repo or code sample that causes the error if relevant

-->

```
on:
  # Trigger analysis when pushing in master or pull requests, and when creating
  # a pull request.
  push:
    branches:
      - master
      - develop
  pull_request:
    types: [opened, synchronize, reopened]

permissions:
  contents: read

name: Bearer SAST scan
jobs:
  rule_check:
    runs-on: default-medium
    steps:
      - uses: actions/checkout@v4
      - name: Install dependencies
        run: sudo apt-get update && sudo apt-get install -y curl
      - name: Run Report
        id: report
        uses: bearer/bearer-action@v2
        with:
          config-file: ".github/bearer.yml"
      - id: summary
        name: Display Summary
        uses: actions/github-script@v6
        with:
          script: |
            // github does not support multiline outputs so report is encoded
            const report = decodeURIComponent(`${{ steps.report.outputs.rule_breaches }}`);
            const passed = `${{ steps.report.outputs.exit_code }}` == "0";
            if(!passed){ core.setFailed(report); }
```

bearer.yml
```
report:
  report: security
  severity: "critical,high,medium"
scan:
  scanner:
    - sast
    - secrets
  skip-path:
    [
      node_modules,
      migrations,
      "*.integration.ts",
      "*.js",
      "*.md",
      "*.patch",
      "*.rb",
      "*.rbi",
      "*.sql",
      "*.test.ts",
      "buildspec-migrations.yml",
      "buildspec.yml",
      ".env*",
      "packages/auth0-app/src/branding.ts",
    ]
rule:
  skip-rule: [
      javascript_lang_exception,
      javascript_express_helmet_missing,
      javascript_lang_weak_hash_md5, # MD5 is fine for our use case
    ]
```

## Expected Behavior

<!--- Tell us what should happen -->
test should run

## Actual Behavior

<!--- Tell us what happens instead -->

```
Analyzing codebase
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x4ffcee]

goroutine 1 [running]:
os.(*Process).signal(0x28?, {0x1717410?, 0x3460728?})
	/usr/local/go/src/os/exec_unix.go:63 +0x2e
os.(*Process).Signal(...)
	/usr/local/go/src/os/exec.go:140
os.(*Process).kill(...)
	/usr/local/go/src/os/exec_posix.go:67
os.(*Process).Kill(...)
	/usr/local/go/src/os/exec.go:1[25](https://github.com/deepcrawl/canary/actions/runs/6969201168/job/18964630752#step:4:28)
os/exec.CommandContext.func1()
	/usr/local/go/src/os/exec/exec.go:449 +0x2d
github.com/bearer/bearer/internal/git.captureCommand({0x171caf0?, 0x35ad9a0?}, {0x7ffe8d29ceeb, 0x1}, {0xc002[28](https://github.com/deepcrawl/canary/actions/runs/6969201168/job/18964630752#step:4:31)68a0?, 0x505c2c?, 0x7ffe8d[29](https://github.com/deepcrawl/canary/actions/runs/6969201168/job/18964630752#step:4:32)ceeb?}, 0xc002286838)
	/github/workspace/internal/git/git.go:94 +0x2aa
github.com/bearer/bearer/internal/git.captureCommandBasic({0x171caf0?, 0x35ad9a0?}, {0x7ffe8d29ceeb?, 0x10c3780?}, {0xc0022868a0?, 0xc001bcd5c0?, 0x10c3780?})
	/github/workspace/internal/git/git.go:114 +0x54
github.com/bearer/bearer/internal/git.GetRoot({0x7ffe8d29ceeb, 0x1})
	/github/workspace/internal/git/git.go:35 +0xa7
github.com/bearer/bearer/internal/commands/process/gitrepository.NewContext(0xc0022874a8)
	/github/workspace/internal/commands/process/gitrepository/context.go:43 +0x45
github.com/bearer/bearer/internal/commands/artifact.Run({_, _}, {{{0x0, 0x0}, {0xc0014186a8, 0x8}, {0x0, 0x0}, 0xc0013dba10, 0xc0013dbad0, ...}, ...})
	/github/workspace/internal/commands/artifact/run.go:289 +0x2f4
github.com/bearer/bearer/internal/commands.NewScanCommand.func2(0xc0012fcf00, {0xc0013db5f0?, 0x1, 0x3})
	/github/workspace/internal/commands/scan.go:90 +0x3f8
github.com/spf13/cobra.(*Command).execute(0xc0012fcf00, {0xc0013db590, 0x3, 0x3})
	/github/home/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:983 +0xabc
github.com/spf13/cobra.(*Command).ExecuteC(0xc0012fc[30](https://github.com/deepcrawl/canary/actions/runs/6969201168/job/18964630752#step:4:33)0)
	/github/home/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1115 +0x3ff
github.com/spf13/cobra.(*Command).Execute(0x170ed00?)
	/github/home/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1039 +0x13
main.run()
	/github/workspace/cmd/bearer/main.go:18 +0x[34](https://github.com/deepcrawl/canary/actions/runs/6969201168/job/18964630752#step:4:37)
main.main()
	/github/workspace/cmd/bearer/main.go:11 +0x13
```

## Possible Fix

<!--- Optional: suggest a fix or reason for the bug if you have a solution -->

## Your Environment

<!--- Include as many relevant details about the environment you experienced the bug in -->

- Operating System and version: linux amd64 (self hosted github runner on k8s)
- Output of 'bearer version':
```
bearer version:  1.31.1
sha: xxx
```

--- a/.goreleaser.yaml
+++ b/.goreleaser.yaml
@@ -14,6 +14,8 @@ brews:
     name: bearer
     folder: Formula
     install: bin.install "bearer"
+    dependencies:
+      - name: git
     tap:
       owner: Bearer
       name: homebrew-tap

--- a/contrib/install.sh
+++ b/contrib/install.sh
@@ -244,6 +244,16 @@ http_copy() {
   echo "$body"
 }
 
+check_dependencies() {
+  log_info "Checking dependencies..."
+  if git --version &> /dev/null; then
+    log_debug "Git is installed. Version: $(git --version)"
+  else
+    log_crit "Git is not installed on this system."
+    return 1
+  fi
+}
+
 github_release() {
   owner_repo=$1
   version=$2
@@ -310,6 +320,7 @@ log_prefix() {
 	echo "$PREFIX"
 }
 
+
 PLATFORM="${OS}/${ARCH}"
 GITHUB_DOWNLOAD=https://github.com/${OWNER}/${REPO}/releases/download
 
@@ -317,6 +328,8 @@ uname_os_check "$OS"
 uname_arch_check "$ARCH"
 
 parse_args "$@"
+
+check_dependencies
 
 get_binaries
 

--- a/internal/git/git.go
+++ b/internal/git/git.go
@@ -91,7 +91,6 @@ func captureCommand(ctx context.Context, workingDir string, args []string, captu
 	}
 
 	if err := command.Start(); err != nil {
-		command.Cancel() //nolint:errcheck
 		return err
 	}
 

